esp-hal/src/rmt.rs

```diff
@@ -710,19 +711,25 @@ for_each_rmt_channel!(
             ];
 
             $(
-                impl<'d, Dm> TxChannelCreator<'d, Dm> for ChannelCreator<Dm, $num>
+                impl<'ch, Dm> TxChannelCreator<'ch, Dm> for ChannelCreator<'ch, Dm, $num>
                 where
                     Dm: crate::DriverMode,
                 {
-                    fn configure_tx(
+                    fn configure_tx<'pin>(
                         self,
-                        pin: impl PeripheralOutput<'d>,
+                        pin: impl PeripheralOutput<'pin>,
                         config: TxChannelConfig,
-                    ) -> Result<Channel<Dm, Tx>, Error>
+                    ) -> Result<Channel<'ch, Dm, Tx>, Error>
```

Comment:
Wouldn't it be simpler to use a single lifetime? What's the advantage of the separate `'pin`?

Reply:
Hmm, I'm not sure now what I was thinking when writing this, I think some concerns regarding what this implies at the call site. However, I can't come with an example now, and I'm not at all sure I understood it properly. Could this save some explicit reborrowing?

I'm pushing a commit to change this to use a single lifetime (which does work for HIL tests and examples). If there's an actual use case where it makes a significant difference, this could still be revisited later.